actors/builtin/power/power_actor.go

```diff
@@ -139,6 +139,12 @@ func (a Actor) CreateMiner(rt Runtime, params *CreateMinerParams) *CreateMinerRe
 
 		st.MinerCount += 1
 
+		// starting version 7, call addToClaim to ensure new claim updates all power stats
+		if rt.NetworkVersion() >= network.Version7 {
+			err := st.updateStatsForNewMiner(params.SealProofType)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed update power stats for new miner %v", addresses.IDAddress)
```

Comment:
Note: In v3 I'd like to push this whole thing including `setClaim` down into a state method.